packages/generator/src/generator.ts

```diff
@@ -55,33 +54,47 @@ const logger = createLogger({
   messageContext: 'generator'
 });
 
-export async function generate(
-  options: GeneratorOptions
-): Promise<void | EmitResult[]> {
+const threadsForCompiling = 16;
+
+export async function generate(options: GeneratorOptions): Promise<void> {
   const project = await generateProject(options);
   if (!project) {
     throw Error('The project is undefined.');
   }
 
   await project.save();
-
   if (options.generateJs) {
-    project
+    let directories = project
       .getDirectories()
-      .filter(dir => !!dir.getSourceFile('tsconfig.json'))
-      .forEach(dir => {
-        logger.info(`Transpiling files in the directory: ${dir.getPath()}.`);
-
-        execa('tsc', { cwd: dir.getPath() }).catch(err => {
-          logger.error(`Error: Failed to generate js files: ${err}`);
-          process.exit(1);
-        });
-      });
+      .filter(dir => !!dir.getSourceFile('tsconfig.json'));
+    while(directories.length > 0){
+      const chunk = directories.splice(0,threadsForCompiling)
+      await transpileDirectories(chunk)
+    }
   }
 
   return;
 }
 
+export async function transpileDirectories(directories:Directory[]):Promise<void[]>{
+  return Promise.all(directories.map(directory=>transpileDirectory(directory)))
+}
+
+export async function transpileDirectory(dir: Directory): Promise<void> {
+  try {
+    logger.info(
+      `Transpiling files in the directory: ${dir.getPath()} started.`
+    );
+    await execa('tsc', { cwd: dir.getPath() })
```

Comment:
Please correct me.
I guess you had the feeling of the difference between these two versions in terms of the performance (CPU usage).
For the code changes, I found this is the line (with `await` keyword) that changes your life and the rest of them is doing the same, including the `16` thing.

Reply:
sorry. I do not really get what you mean.

Reply:
Discussed, `await` will be removed.